src/test/java/org/apache/commons/collections4/bloomfilter/LayerManagerTest.java

```diff
@@ -0,0 +1,305 @@
+    }
+
+    @Test
+    public void testCopy() {
+        LayerManager underTest = LayerManager.builder().setSupplier(() -> new SimpleBloomFilter(shape)).build();
+        underTest.getTarget().merge(TestingHashers.randomHasher());
+        underTest.next();
+        underTest.getTarget().merge(TestingHashers.randomHasher());
+        underTest.next();
+        underTest.getTarget().merge(TestingHashers.randomHasher());
+        assertEquals(3, underTest.getDepth());
+
+        LayerManager copy = underTest.copy();
+        assertNotSame(underTest, copy);
+        // object equals not implemented
+        assertNotEquals(underTest, copy);
+
+        assertEquals(underTest.getDepth(), copy.getDepth());
+        assertTrue(
+                underTest.forEachBloomFilterPair(copy, (x, y) -> Arrays.equals(x.asBitMapArray(), y.asBitMapArray())));
+    }
+
+    @Test
+    public void testBuilder() {
+        LayerManager.Builder underTest = LayerManager.builder();
+        NullPointerException npe = assertThrows(NullPointerException.class, () -> underTest.build());
+        assertTrue(npe.getMessage().contains("Supplier must not be null"));
+        underTest.setSupplier(() -> null).setCleanup(null);
+        npe = assertThrows(NullPointerException.class, () -> underTest.build());
+        assertTrue(npe.getMessage().contains("Cleanup must not be null"));
+        underTest.setCleanup(x -> {
+        }).setExtendCheck(null);
+        npe = assertThrows(NullPointerException.class, () -> underTest.build());
+        assertTrue(npe.getMessage().contains("ExtendCheck must not be null"));
+
+        npe = assertThrows(NullPointerException.class, () -> LayerManager.builder().setSupplier(() -> null).build());
+        assertTrue(npe.getMessage().contains("filterSupplier returned null."));
+
+    }
+
+    @Test
+    public void testClear() {
+        LayerManager underTest = LayerManager.builder().setSupplier(() -> new SimpleBloomFilter(shape)).build();
+        underTest.getTarget().merge(TestingHashers.randomHasher());
+        underTest.next();
+        underTest.getTarget().merge(TestingHashers.randomHasher());
+        underTest.next();
+        underTest.getTarget().merge(TestingHashers.randomHasher());
+        assertEquals(3, underTest.getDepth());
+        underTest.clear();
+        assertEquals(1, underTest.getDepth());
+        assertEquals(0, underTest.getTarget().cardinality());
+    }
+
+    @Test
+    public void testNextAndGetDepth() {
+        LayerManager underTest = LayerManager.builder().setSupplier(() -> new SimpleBloomFilter(shape)).build();
+        assertEquals(1, underTest.getDepth());
+        underTest.getTarget().merge(TestingHashers.randomHasher());
+        assertEquals(1, underTest.getDepth());
+        underTest.next();
+        assertEquals(2, underTest.getDepth());
+    }
+
+    @Test
+    public void testGet() {
+        LayerManager underTest = LayerManager.builder().setSupplier(() -> new SimpleBloomFilter(shape)).build();
```

Comment:
The SimpleBloomFilter could be cached and then returned. You can then assert that the LayerManager either does or does not return a copy:
```java
SimpleBloomFilter f = new SimpleBloomFilter(shape);
LayerManager underTest = LayerManager.builder().setSupplier(() -> f).build()
// ...
assertSame(f, underTest.get(0));
```